decepticonlp/attacks/attack.py

```diff
@@ -0,0 +1,328 @@
+from tqdm import tqdm
+import torch
+
+loss_type_list = [
+    torch.nn.modules.loss.BCELoss,
+    torch.nn.modules.loss.BCEWithLogitsLoss,
+    torch.nn.modules.loss.CrossEntropyLoss,
+    torch.nn.modules.loss.NLLLoss,
+]
+
+
+class CharAttackerWithHuggingFace:
+    """
+		Args:
+		model: PyTorch model
+			-Pretrained PyTorch model (trained on benign training set) for inference
+		
+		data_loader: torch.utils.data.Dataloader 
+			-Dataloader object of the original test set
+		
+		adversarial_data_loader: torch.utils.data.Dataloader
+			-Dataloader object of the adversarial test set
+		
+		input_format: list (of strings) (default: ['input_ids','labels'])
+			-The input format as you would feed to your model.
+			Eg:
+				For BERTForSequenceClassification, the input_format will be ['input_ids','attention_masks','labels'] (if you are feeding only these three to the model).
+		
+		accuracy: boolean (default: True)
+			-If True, logs and prints the accuracy
```

Comment:
Also what docstring format is this?

Reply:
https://github.com/pytorch/vision/blob/master/torchvision/transforms/transforms.py
This is what I followed